### PR TITLE
dm-syncer support route-rules

### LINKED ISSUE #599: dm-syncer support route-rules
## Feature Request

**Is your feature request related to a problem? Please describe:**

dm-syncer is a tools compiled from dm's syncer unit, and is used to replace syncer. issue: https://github.com/pingcap/dm/issues/409

dm-syncer already support config `route-rules`, but we have not to check whether this config can work.

**Describe the feature you'd like:**

1. add test case for `route-rules` in dm-syncer's integration test
2. if dm-syncer's `route-rules` can not work, need to fix it

## Score (Not for Hacktoberfest)

200

## Mentor(s)
* @csuzhangxc 

Contact the mentors: **#tidb-challenge-program** channel in [TiDB Community](https://join.slack.com/t/tidbcommunity/shared_invite/enQtNzc0MzI4ODExMDc4LWYwYmIzMjZkYzJiNDUxMmZlN2FiMGJkZjAyMzQ5NGU0NGY0NzI3NTYwMjAyNGQ1N2I2ZjAxNzc1OGUwYWM0NzE)  Slack Workspace

For Hacktoberfest: **#sig-migrate** in [TiDB Community](https://slack.tidb.io/invite?team=tidb-community&channel=sig-migrate&ref=github_sig) Slack Workspace

## Recommended Skills

* Golang
* Experience with DM

## Learning Materials

- [DM document](https://docs.pingcap.com/tidb-data-migration/stable/overview)
- [Syncer document](https://docs.pingcap.com/tidb/stable/syncer-overview)
- [DM-syncer's integration test](https://github.com/pingcap/dm/tree/master/tests/dm_syncer)

### PR DESCRIPTION
**Pull request by Howie59**
Signed-off-by: lihaowei <haoweili35@gmail.com>

### What problem does this PR solve? 
- close #599 

### Check List 

Tests 
 - Integration test

Side effects
 - Increased code complexity

### CODE REVIEW

File: tests/dm_syncer/conf/diff_config_route_rules.toml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+fix-sql-file = "fix.sql"
+
+# tables need to check.
+[[check-tables]]
+schema = "dm_syncer"
+tables = ["dm_syncer_route_rule"]
+
+[[table-config]]
+schema = "dm_syncer"
+table = "dm_syncer_route_rule_1"
```
Comment (lance6716) [inline tests/dm_syncer/conf/diff_config_route_rules.toml:24]: please follow https://docs.pingcap.com/zh/tidb/stable/shard-diff

`table` here should be downstream "merged" table, and should set `is-sharding = true`

**Comment by lance6716** [comment]
Hi, you could see there's a error message in [fail CI](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7278/pipeline), which means source tables in a route group can't have a auto-increment key.

**Comment by lance6716** [comment]
3 source tables have conflict in PK if merged, thus there is an error. please use different value

**Comment by lance6716** [comment]
error is caused by `diff_config,toml`, because downstream table `dm_syncer_route_rules` also matches the regular expression `t.*` so this table is about to check. I guess after your addition, this re should be
```
tables = ["~^t.*"]
```

**Comment by Howie59** [comment]
> error is caused by `diff_config,toml`, because downstream table `dm_syncer_route_rules` also matches the regular expression `t.*` so this table is about to check. I guess after your addition, this re should be
> 
> ```
> tables = ["~^t.*"]
> ```

oh, my mistake, thx

**Comment by lance6716** [comment]
LGTM

**Review by csuzhangxc — APPROVED** [review]
LGTM

**Pull request merged**